src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -953,6 +953,7 @@ def build_proto_from_callable(
     expr_builder: proto.SpCallable,
     func: Union[Callable, Tuple[str, str]],
     ast_batch: Optional[AstBatch] = None,
+    object_name: Optional[str] = None,
```

Comment:
we may want to type it with the weird `Optional[Union[str, List[str]]]` that is used across snowpark to identify objects.